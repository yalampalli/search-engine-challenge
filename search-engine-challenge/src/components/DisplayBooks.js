import React from "react";

export const Book = ({ book }) => {
  return (
    <div className="book">
      <div data-testid="book-title" className="title">
        {book.title}
      </div>
      <div data-testid="book-summary" className="summary">
        {book.summary}
      </div>
      <h4 data-testid="book-author">
        <hr />
        {book.author.author}
      </h4>
    </div>
  );
};

const DisplayBooks = ({ books }) => {
  return (
    <div data-testid="book-container" className="bookContainer">
      {books.map(book => (
        <Book key={book.title} book={book} />
      ))}
    </div>
  );
};

export default DisplayBooks;
