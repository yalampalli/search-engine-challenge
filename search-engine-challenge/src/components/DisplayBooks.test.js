import React from "react";
import { render } from "@testing-library/react";
import DisplayBooks, { Book } from "./DisplayBooks.js";

const books = [
  {
    title: "Title 1",
    summary: "Summary 1",
    author: { author: "Author 1" }
  },
  {
    title: "Title 2",
    summary: "Summary 2",
    author: { author: "Author 2" }
  }
];

describe("Display books Component", () => {
  it("should render correct number of books", async () => {
    const { getByTestId } = render(<DisplayBooks books={books} />);
    const elem = getByTestId("book-container");

    expect(elem.children.length).toBe(2);
  });

  it("should match the snapshot of the book for same input", () => {
    const { container } = render(<Book book={books[0]} />);
    expect(container).toMatchSnapshot();
  });

  it("should render title, summary and author in the book component", () => {
    const { getByTestId } = render(<Book book={books[0]} />);

    const title = getByTestId("book-title");
    expect(title.textContent).toBe("Title 1");

    const summary = getByTestId("book-summary");
    expect(summary.textContent).toBe("Summary 1");

    const author = getByTestId("book-author");
    expect(author.textContent).toBe("Author 1");
  });
});
