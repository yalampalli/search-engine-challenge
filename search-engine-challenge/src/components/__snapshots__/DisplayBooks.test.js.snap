// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display books Component should match the snapshot of the book for same input 1`] = `
<div>
  <div
    class="book"
  >
    <div
      class="title"
      data-testid="book-title"
    >
      Title 1
    </div>
    <div
      class="summary"
      data-testid="book-summary"
    >
      Summary 1
    </div>
    <h4
      data-testid="book-author"
    >
      <hr />
      Author 1
    </h4>
  </div>
</div>
`;
